### PR TITLE
Module/iam  enum_action_query

### PR DESCRIPTION
**Pull request by DaveYesland**
This adds a new module which allows you to query permissions/actions from enumerated permissions.

### CODE REVIEW

File: pacu/modules/iam__enum_action_query/main.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    "category": "ENUM",
+    "one_liner": "Allows you to query enumerated user and role permissions.",
+    "description": "This module allows you to query IAM permissions for users and roles and see what resources if any they have those permissions on. For example --query s3:get*,iam:create*.", # noqa
+    "services": ["IAM"],
+    "prerequisite_modules": ["iam__enum_permissions"],
+    "external_dependencies": [],
+    "arguments_to_autocomplete": [],
+}
+
+parser = argparse.ArgumentParser(add_help=False, description=module_info["description"])
```
Comment (jdearmas) [inline pacu/modules/iam__enum_action_query/main.py:23]: `add_help=False`

You don't want to add a help flag?

**Review by jdearmas — APPROVED** [review]
This looks good. I'll list a few nice-to-haves:

- exception handling
- a test or two

File: pacu/modules/iam__enum_action_query/main.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python3
```
[the remark this answers is not in the document]
Comment (DaveYesland) [reply pacu/modules/iam__enum_action_query/main.py:1]: I feel this is handled in the module_info at the top.

File: pacu/modules/iam__enum_action_query/main.py (hunk 1 above)
Comment (DaveYesland) [reply pacu/modules/iam__enum_action_query/main.py:23]: Fixed

File: pacu/modules/iam__enum_action_query/main.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+            return f"{Fore.YELLOW}{string}{Style.RESET_ALL}"
+
+    if args.folder:
+        iam_enum_folder = args.folder
+    else:
+        iam_enum_folder = "{}/confirmed_permissions/".format(downloads_dir())
+
+    if os.path.isdir(iam_enum_folder) is False:
+        print(
+            f'{iam_enum_folder} not found! Maybe you have not run {module_info["prerequisite_modules"][0]} yet...\n'
```
[the remark this answers is not in the document]
Comment (DaveYesland) [reply pacu/modules/iam__enum_action_query/main.py:104]: Was a copy and paste but good call, fixed.

File: pacu/modules/iam__enum_action_query/main.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+    # List all the files in the iam_query_data directory
+    # which is created by the iam__enum_permissions module
+    files = os.listdir(iam_enum_folder)
+
+    # If the user has specified a role or user to filter by
+    # then only use the files that match that role or user
+    if args.role:
+        files = [file for file in files if f"role-{args.role}.json" in file]
+    elif args.user:
```
[the remark this answers is not in the document]
Comment (DaveYesland) [reply pacu/modules/iam__enum_action_query/main.py:125]: fixed

**Pull request merged**